Makefile.PL

```diff
@@ -1,62 +1,99 @@
 use strict;
 use warnings;
 use ExtUtils::MakeMaker;
+use Devel::CheckLib;
+use Config ();
+
+# depending on the compiler name as defined in %Config::Config,
+# check for the library, head, and flags needed to use OpenMP
+if ( q{gcc} eq $Config::Config{ccname} ) {
+    local $/ = undef;          # slurp mode for main() body in __DATA__
+    check_lib_or_exit(
+        lib     => q{gomp},
+        header  => q{omp.h},
+        ccflags => q{-fopenmp},
+        ldflags => q{-fopenmp},
+        function => <DATA>,    # main() body defined in __DATA__
+    );
```

Comment:
This is still not a `%check_lib_args` defined above the `if` block, modified within the `if` block, and one `check_lib_or_exit` call in the whole program. What would the code look like if you supported a second environment?

Reply:
Oh, you're saying to set it up to be easily extended in the future.